### ISSUE TITLE
Enterprise repo picker follow up polish

### ISSUE THREAD

**Issue by dominiccooney** [sourcegraph/jetbrains#1322]
These are follow up issues from #982 and code review feedback on #1308

```[tasklist]
### Tasks
- [ ] Add [tooltip](https://jetbrains.design/intellij/controls/tooltip/) on hover of "Chat Context" for enterprise users with below text.
- [ ] Remove the "Help" toolbar icon, make the toolbar icon one entry tall (at least in Enterprise), and make the minimum size smaller.
- [x] Dismiss the repo selection window when clicking outside of it
- [ ] The cmd-enter to accept language is small and low contrast, but we are using the theme colors; consider using autosave on close @danielmarquespt
- [ ] Add "Cody Context" entry as part of the [Search Everywhere](https://www.jetbrains.com/help/idea/searching-everywhere.html)
- [ ] Add "Cody Context" as part of the Keymap list on JetBrains settings
- [ ] Try to switch to Action
- [ ] Make tree view reflect language below
- [ ] Show the "inferred" repo, exact UX TBD @danielmarquespt
- [ ] Consider switching to ActionManager.createActionToolbar to get more toolbar padding like the git log viewer. Note sporadic problem with the toolbar button hot zone being the right hand side of the button.
- [ ] When individual repos are un/checked, reflect that in the count in the summary line ("Chat Context X repos ...")
- [ ] #1318
- [ ] Make it accept full URLs and strip the https://
```

Tooltip text:

```
About chat context
Select up to 10 repositories from [customer_instance_url] to use as reference context in this chat.
 
[Context Docs](https://sourcegraph.com/docs/cody/core-concepts/context)
```

Tree view language:

```
Chat Context X repos on [instance]
    Repositories
         example/example
         example/example2
```

**Comment by danielmarquespt** [comment]
Added `Make it accept full URLs and strip the https://` as a user just faced this problem and was unsure why the repo was not being accepted

**Comment by kalanchan** [comment]
waiting on review

**Comment by toolmantim** [comment]
I've jumped in here and done a review at the request of @dominiccooney so he can get started on the problems that were raised in the call last night, as he needed to get started on the eng side. Apologies for the dump here, just wanting to get them out quickly…

---

1. The treeview items should match what’s added in the edit textarea 1:1 (aside from duplicates)
    - e.g. if you add “dev.azure.com/sourcegraph-mike/Test Project/repo1“ on s2, it doesn’t appear in the treeview at all with no feedback as to why
    - Instead, repos that aren’t valid are shown as disabled in the treeview, with a small message as to why they are ignored

Currently: 

![CleanShot 2024-05-17 at 12 45 42@2x](https://github.com/sourcegraph/jetbrains/assets/153/4c975dc8-46ca-4f27-a814-c9c963cb1dee)

![CleanShot 2024-05-17 at 12 45 55@2x](https://github.com/sourcegraph/jetbrains/assets/153/3c3b1916-ffc2-460d-9e81-f8655fcd3f86)

Instead here is an updated design that shows a few tweaks things labels, unnecessary nestings, unnecessary server name etc, as well as the invalid repo:

https://www.figma.com/design/m9Yvgv6W3aPRLfXnxgwnEz/JetBrains?node-id=1540-6065&t=tnA1l77M1NjRHLQh-4

<img src="https://github.com/sourcegraph/jetbrains/assets/153/95bb3d45-b06f-44d7-9668-d201bf24e71f" width="507"/>

---

2. If the list is edited (e.g. the list is reordered, a new one is added), we retain the unchecked state of any repos in the treeview (I.e. it doesn’t just reset them all to be checked).
    - Removing a repo and re-adding it as two steps doesn’t need to remember the unchecked state, it should go back to being checked.

---

3. Position the context tooltip above the treeview section, never overlapping it

---

4. Nit 1: Clicking inside the input below the last line should put caret at the end of the last char on the last line

https://github.com/sourcegraph/jetbrains/assets/153/96a4f9ba-0465-496f-8270-13d76068cf1a

---

5. Nit 2: More top and bottom padding around the treeview so it's centred 

![CleanShot 2024-05-17 at 14 22 39@2x](https://github.com/sourcegraph/jetbrains/assets/153/dc976384-7c0b-4942-a0d0-ee6a94a39f0c)

---

Longer-term I think it's probably worth asking: is this the right type of UI here, or should we investigate a probably-more-easy-to-use dialog for doing this… something more like:

![CleanShot 2024-05-17 at 14 06 35@2x](https://github.com/sourcegraph/jetbrains/assets/153/57fa0aec-0801-40d8-8fef-648a4a9de640)

Be curious to see how people feel about the current UI, and @danielmarquespt we could work on a different version of it.

**Comment by danielmarquespt** [comment]
thank you @toolmantim for jumping in on this.

I agree with the things you suggested, but I'll be conducting a review myself and share (other?) things that I might find, as well as the priority for them

**Comment by danielmarquespt** [comment]
### List of improvements by priority
(contains reviewed suggestions from @toolmantim and more)


- [ ] **Rearrange the tree view to this:**

<img width="574" alt="Screenshot 2024-05-17 at 17 51 45" src="https://github.com/sourcegraph/jetbrains/assets/7814431/1efad798-547b-45b9-9799-9a95f7f3f3b9">

1st line enables or disables all of the checkboxes below them
The inferred repo has the checkbox disabled
The counter of selected repos from [instance] moves from the first line to the node that represents external repos

---

- [ ] **With no extra repos added we display a blank stale message**

<img width="625" alt="Screenshot 2024-05-17 at 17 53 44" src="https://github.com/sourcegraph/jetbrains/assets/7814431/d3965c7c-584f-4fe1-a2d3-b14fab0e054f">

`Use the pencil ✎ icon to add and remove repos ↗`

This message should be a tree node like the other items, without icon, chevron, checkbox, and using the hint style

---

- [ ] **There is 1 to 1 parity between the entries on both**

Removing an item from the text box, removes it from the tree. Used to be always true, but I'm seeing weird states there tree nodes still pressist after being removed from the text box 

---

- [ ] **Warnings are applied to both the box and the list**

<img width="1122" alt="Screenshot 2024-05-17 at 17 59 55" src="https://github.com/sourcegraph/jetbrains/assets/7814431/87e27fe2-c55a-43da-8b9d-ea047997d2a1">

Invalid repos display the full string on the tree (since can be some gibberish or invalid urls, we can’t format it properly, so we can display that string with a ? icon)

---

- [ ] **Invalid repos display a warning (used to be the case but not seeing it anymore)**

<img width="461" alt="Screenshot 2024-05-17 at 17 58 03" src="https://github.com/sourcegraph/jetbrains/assets/7814431/4dab164d-ca7a-452e-9a33-a10008d1495e">

---

- [ ] **Edit box needs to accepts full urls**

like `https://github.com/sourcegraph/jetbrains/` → adds to the tree as `sourcegraph/jetbrains`

---

- [ ] **Tool tips needs to be a bit farther from the sidebar. 30px seems good**

<img width="616" alt="Screenshot 2024-05-17 at 18 04 00" src="https://github.com/sourcegraph/jetbrains/assets/7814431/6956a8a3-2791-4770-a1fa-3b3054c79789">

---

- [ ] **Retain the unchecked state of any repos**

If the list is edited (e.g. the list is reordered, a new one is added), we retain the unchecked state of any repos in the treeview (I.e. it doesn’t just reset them all to be checked).
Removing a repo and re-adding it as two steps doesn’t need to remember the unchecked state, it should go back to being checked.

---

- [ ] **Clicking inside the input below the last line should put caret at the end of the last char on the last line**

https://github.com/sourcegraph/jetbrains/assets/153/96a4f9ba-0465-496f-8270-13d76068cf1a

---

(other layout and spacing coming as a different issue. Functionality first)

**Comment by danielmarquespt** [comment]
### List of improvements by priority UPDATED 21/05
(contains reviewed suggestions from @toolmantim and more)


- [ ] **Rearrange the tree view to this:**

<img width="657" alt="Screenshot 2024-05-21 at 15 04 51" src="https://github.com/sourcegraph/jetbrains/assets/7814431/89b49f48-de99-4cc0-922e-7dbf0d788a84">

Needed icons:
[icons_reposelector.zip](https://github.com/sourcegraph/jetbrains/files/15390248/icons_reposelector.zip)

1st line enables or disables all of the checkboxes below them
The inferred repo has the checkbox disabled
The counter of selected repos from [instance] moves from the first line to the node that represents external repos

---

- [ ] **Move the edit action from the sidebar to the last line of the tree**

---

- [ ] **Add "Type to add repos from sourcegraph.sourcegraph.com" to the repo text area**

<img width="687" alt="Screenshot 2024-05-21 at 14 54 20" src="https://github.com/sourcegraph/jetbrains/assets/7814431/1c229698-1acf-4dfd-8884-ea76b72f2309">

---

- [ ] **Add a import com.intellij.ui.TitledSeparator to the top of the repo selector**

<img width="890" alt="Screenshot 2024-05-21 at 14 52 11" src="https://github.com/sourcegraph/jetbrains/assets/7814431/c78e0fd9-e07e-4c38-96b5-d40a2885293a">

string: Chat Context Settings

---

- [ ] **Remove tooltip from treeview and move it to the "Chat Context Settings" separator (on hover)**

---

- [ ] **There is 1 to 1 parity between the entries on both**

Removing an item from the text box, removes it from the tree. Used to be always true, but I'm seeing weird states there tree nodes still pressist after being removed from the text box 

---

- [ ] **Warnings are applied to both the box and the list**

<img width="1122" alt="Screenshot 2024-05-17 at 17 59 55" src="https://github.com/sourcegraph/jetbrains/assets/7814431/87e27fe2-c55a-43da-8b9d-ea047997d2a1">

Invalid repos display the full string on the tree (since can be some gibberish or invalid urls, we can’t format it properly, so we can display that string with a ? icon)

---

- [ ] **Edit box needs to accepts full urls**

like `https://github.com/sourcegraph/jetbrains/` → adds to the tree as `sourcegraph/jetbrains`

---

- [ ] **Retain the unchecked state of any repos**

If the list is edited (e.g. the list is reordered, a new one is added), we retain the unchecked state of any repos in the treeview (I.e. it doesn’t just reset them all to be checked).
Removing a repo and re-adding it as two steps doesn’t need to remember the unchecked state, it should go back to being checked.

---

- [ ] **Clicking inside the input below the last line should put caret at the end of the last char on the last line**

https://github.com/sourcegraph/jetbrains/assets/153/96a4f9ba-0465-496f-8270-13d76068cf1a

---

(UI details and fixes coming in a different issue. Functionality first)

**Comment by toolmantim** [comment]
@danielmarquespt "Local project" is a problematic label, because although the repo is detected from the local project, the code is all coming from the remote repo… no local git changes are included (which is a problem/surprise for a lot of users currently… this could make it worse). I think "Project repository" is probably a less problematic label.

Given the above… and that all the repos are from the instance… I think we can do without the Enterprise instance URL in the treeview. Maybe in the edit window is the right place to explain they can only add repos that exist on their instance?

Given all that… I think we can remove the extra level of tree nesting ([Figma](https://www.figma.com/design/m9Yvgv6W3aPRLfXnxgwnEz/JetBrains?node-id=1540-6065&t=EFaxdAYiwJx0kBOL-4)):

I'd suggest something like this for initial state: _(last item opens edit view when clicked)_

![image](https://github.com/sourcegraph/jetbrains/assets/153/e4ba1201-425a-4ce1-bcf1-cccf80d207f1)

If > 1 extra repo: _(last item opens edit view when clicked)_

![image](https://github.com/sourcegraph/jetbrains/assets/153/f492f65b-abef-4ccd-b12b-59b4cecc7c99)

Collapsed view showing a count and some state, so when it's collapsed you can still tell what the scope of the chat is:

![image](https://github.com/sourcegraph/jetbrains/assets/153/9b0aad03-e525-4b7e-8312-214a9538781c)

**Comment by danielmarquespt** [comment]
Ok I just tested because I was though that the source of the local project was the local files. I tested with a local project that is not part of the sourcegraph instance and Cody kinda works, and is able to fecth files but its way more limited.

Given that I think it makes sense to put the local project under the instance node, and we can simplify the tree. 

I'll update this

---

Regarding having the action at the bottom of the tree, I like it, but as far as I understand we cannot include actions as part of tree nodes. Maybe we can listen for a click there and trigger the edit box, but I'm not sure of much of an hack that would be

**Comment by toolmantim** [comment]
If it's anything like VS Code it'll have a click handler you can trigger anything off.

**Comment by dominiccooney** [comment]
There's multiple lists here with some conflicting details (Repos? Repositories? ... Say the remote name? Don't say the remote name?) and some of the list items are duplicates (for example #1354 tracks supporting https clone URLs.)

I am making a clean up pass in dpc/issue-1322 with these changes (so far):

- Add the pencil/plus item, with different language for add vs edit, and have it trigger the popup.
- Make the tree view single selection, which makes clicking on the add/edit more natural.
- Fix the shade of icons; add dark variants.
- Add the "Project repository" label on the automatically included repository.
- Add the "Ignored by admin" label on the automatically included repository.
- Simplify the counters to just "1 active repository", etc.
- Position the popup above the repo list, instead of obscuring it (#1554)
- Endpoint name in the repo editor popup


TODO:
- Temporarily back to uncheck to remove, while I work on the model.

**Comment by toolmantim** [comment]
@dominiccooney so @danielmarquespt and I just caught up, and Figma now has the latest changes with some simplifications.

**Comment by danielmarquespt** [comment]
Updated the task list https://github.com/sourcegraph/jetbrains/issues/1322#issuecomment-2117937246

**Comment by dominiccooney** [comment]
@danielmarquespt , bit of feedback based on the design I had been implementing: the "active" language looks weird to me when enhanced context is unchecked. The repositories aren't "active" in any sense then:

![Screenshot 2024-05-23 at 8 26 42](https://github.com/sourcegraph/jetbrains/assets/55120/07d10a55-0cc5-4835-ad0a-ee6c549f3f5e)

@toolmantim , when you say "the" Figma is that the link in [this comment](https://github.com/sourcegraph/jetbrains/issues/1322#issuecomment-2119749766) and [this comment](https://github.com/sourcegraph/jetbrains/issues/1322#issuecomment-2116602663) ... that still has some $placeholders.

**Comment by toolmantim** [comment]
I did a search in the Figma for those strings, and found https://www.figma.com/design/m9Yvgv6W3aPRLfXnxgwnEz/JetBrains?node-id=1562-1092&m=dev which I think might be the current version of @danielmarquespt’s designs

**Comment by dominiccooney** [comment]
Thanks @toolmantim , noted... next time I will search for the strings. Did not think of that!

**Comment by danielmarquespt** [comment]
Thats a good point. I think we can remove the active and go with just "☑️ Enhanced Context 3 repos"
